src/Compilers/Core/Portable/RuleSet/RuleSetInclude.cs

```diff
@@ -77,42 +77,53 @@ public RuleSet LoadRuleSet(RuleSet parent)
         /// <param name="parent">The parent of this rule set include</param>
         private string GetIncludePath(RuleSet parent)
         {
-            List<string> found = new List<string>();
-            string expandedPath = PortableShim.Environment.ExpandEnvironmentVariables(_includePath);
+            string expandedIncludePath = PortableShim.Environment.ExpandEnvironmentVariables(_includePath);
+            var parentRulesetPath = parent?.FilePath;
 
-            if (PathUtilities.IsUnixLikePlatform)
+            var resolvedIncludePath = ResolveIncludePath(expandedIncludePath, parentRulesetPath);
+            if (resolvedIncludePath == null && PathUtilities.IsUnixLikePlatform)
             {
-                expandedPath = expandedPath.Replace('\\', PortableShim.Path.DirectorySeparatorChar);
+                // Attempt to resolve legacy ruleset includes by replacing Windows style directory separator char with current plaform's directory separator char.
+                resolvedIncludePath = ResolveIncludePath(expandedIncludePath, parentRulesetPath, replaceDirectorySeparatorChar: true);
```

Comment:
Rather than replacing the separators in the expanded path I'd replace them on the original string and then expand the variables again. In that way we won't replace the separators in the strings inserted by the expansion.